File: frontend/src/api/translation.ts
import { api } from './client';
import type { TranslationResult } from '../types/translation';

export const fetchTranslation = (
  fieldName: string,
  jurisdictionId: string,
) =>
  api.post<TranslationResult>('/translation/explain', {
    field_name: fieldName,
    jurisdiction_id: jurisdictionId,
  });

File: src/api/routers/translation.py
from __future__ import annotations

import logging

from fastapi import APIRouter

from ..schemas.translation import TranslationRequest, TranslationResult
from ..services.translation_service import translation_service

logger = logging.getLogger(__name__)
router = APIRouter(prefix="/api/translation", tags=["translation"])


@router.post("/explain", response_model=TranslationResult)
def explain_field(req: TranslationRequest):
    logger.info(
        "POST /translation/explain: field='%s' jid='%s'",
        req.field_name, req.jurisdiction_id,
    )
    return translation_service.explain(
        field_name=req.field_name,
        jurisdiction_id=req.jurisdiction_id,
        code_snippet=req.code_snippet,
        xpaths=req.xpaths,
        dependencies=req.dependencies,
    )
